edx_api/grades/models.py

```diff
@@ -1,23 +1,69 @@
 """
 Business objects for the Grades API
 """
-from collections import Iterable
 
-from six import python_2_unicode_compatible
+from __future__ import unicode_literals
+from six import PY2, python_2_unicode_compatible
 
+# pylint: disable=no-name-in-module, import-error
+if PY2:
+    from collections import Iterable
+else:
+    from collections.abc import Iterable
+# pylint: enable=no-name-in-module, import-error
 # pylint: disable=too-few-public-methods
 
 
-@python_2_unicode_compatible
 class CurrentGrades(object):
     """
     Current Grades object representation
     """
+    @property
+    def all_current_grades(self):
+        """Helper property to return all the CurrentGrade objects"""
+        return self.current_grades.values()  # pylint: disable=no-member
+
+
+@python_2_unicode_compatible
+class CurrentGradesByCourse(CurrentGrades):
+    """
+    Represents the current grades for a specific course
+    """
     def __init__(self, current_grade_list):
         if not isinstance(current_grade_list, Iterable):
-            raise TypeError('CurrentGrades needs a Iterable object')
+            raise TypeError('CurrentGrades needs an Iterable object')
+        self.course_id = None
         self.current_grades = {}
```

Comment:
Nit: could we define it in `init` of the superclass and call `super` at the beginning of the `init` here?
The `# pylint: disable=no-member` would also no longer be needed.

Reply:
Sure @Agrendalath, I generalized the __init__ function so it works for both classes using variable names to select the attributes. This adds some complexity that makes it a bit harder to read, but it definitely de-duplicates a lot of code. Should we switch back to something in the middle where there's some code duplication? Or maybe I can just add some extra comments to make it a bit clearer?

Reply:
Thanks, @kaizoku!
It looks nice, but it's a bit hard to read this without comments. Could you please document the specific steps in the superclass? A docstring explaining args would also be really handy.

Reply:
Yeah this definitely makes it a bit more confusing.
I added some comments to `CurrentGrades.__init__()`, as well as a docstring.
Does that make it a bit better?

Reply:
Yep, looks good!